cherab/solps/mesh_geometry.py

```diff
@@ -302,6 +329,26 @@ def plot_mesh(self):
             patches.append(Polygon(vertices, closed=True))
         p = PatchCollection(patches, facecolors='none', edgecolors='b')
         ax.add_collection(p)
-        ax.axis('equal')
+        ax.set_aspect(1)
+        ax.set_xlim(self.mesh_extent["minr"], self.mesh_extent["maxr"])
+        ax.set_ylim(self.mesh_extent["minz"], self.mesh_extent["maxz"])
+
+        return ax
+    
+    def plot_quadrangle_mesh(self):
+        """
+        Plot the quadrangle mesh grid geometry to a matplotlib figure.
+        """
+        fig, ax = plt.subplots()
+        patches = []
+        for quadrangle in self.quadrangles:
+            vertices = self.vertex_coordinates[quadrangle]
+            patches.append(Polygon(vertices, closed=True))
+        p = PatchCollection(patches, facecolors='none', edgecolors='b')
+        ax.add_collection(p)
+        ax.set_aspect(1)
+        ax.set_xlim(self.mesh_extent["minr"], self.mesh_extent["maxr"])
+        ax.set_ylim(self.mesh_extent["minz"], self.mesh_extent["maxz"])
 
```

Comment:
These two function can be simplified by using `PolyCollection` instead of `PatchCollection` but also expanded in functionality by adding additional argument `solps_data`. I propose the following (optional) change:
```python
    def plot_triangle_mesh(self, solps_data=None):
        """
        Plot the triangle mesh grid geometry to a matplotlib figure.

        :param solps_data: Data array defined on the SOLPS mesh
        """
        fig, ax = plt.subplots()
        verts = self.vertex_coordinates[self.triangles]
        if solps_data is None:
            collection_mesh = PolyCollection(verts, facecolor="none", edgecolor='b', linewidth=0.5)
        else:
            collection_mesh = PolyCollection(verts)
            collection_mesh.set_array(solps_data[self.triangle_to_grid_map[:, 0], self.triangle_to_grid_map[:, 1]])
        ax.add_collection(collection_mesh)
        ax.set_aspect(1)
        ax.set_xlim(self.mesh_extent["minr"], self.mesh_extent["maxr"])
        ax.set_ylim(self.mesh_extent["minz"], self.mesh_extent["maxz"])
        ax.set_ylabel("R [m]")
        ax.set_ylabel("Z [m]")

        return ax

    def plot_quadrangle_mesh(self, solps_data=None):
        """
        Plot the quadrangle mesh grid geometry to a matplotlib figure.

        :param solps_data: Data array defined on the SOLPS mesh
        """
        fig, ax = plt.subplots()
        verts = self.vertex_coordinates[self.quadrangles]
        if solps_data is None:
            collection_mesh = PolyCollection(verts, facecolor="none", edgecolor='b', linewidth=0.5)
        else:
            collection_mesh = PolyCollection(verts)
            collection_mesh.set_array(solps_data[self.quadrangle_to_grid_map[:, 0], self.quadrangle_to_grid_map[:, 1]])
        ax.add_collection(collection_mesh)
        ax.set_aspect(1)
        ax.set_xlim(self.mesh_extent["minr"], self.mesh_extent["maxr"])
        ax.set_ylim(self.mesh_extent["minz"], self.mesh_extent["maxz"])
        ax.set_ylabel("R [m]")
        ax.set_ylabel("Z [m]")

        return ax
```

Reply:
This is a good point. I applied the proposed changes and also added the ax as an optional parameter, so the methods can be used to add the mesh to externally created plots.